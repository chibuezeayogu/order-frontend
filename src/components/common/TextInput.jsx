import React from 'react'

const TextInput = ({ value, onChange, text }) => {
  return (
    <input
      type="text"
      className="form-control rounded-left"
      value={value}
      onChange={onChange}
      id="email"
      placeholder={text}
    />
  )
}

export default TextInput
