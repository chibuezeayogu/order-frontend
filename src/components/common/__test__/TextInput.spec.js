import React from 'react'
import { shallow } from 'enzyme'
import TextInput from '../TextInput'

const setUp = () => {
  const props = {
    value: '',
    onChange: jest.fn(),
    text: 'email'
  }

  return shallow(<TextInput {...props} />)
}

describe('TextInput Component', () => {
  it('should render without crashing', () => {
    const wrapper = setUp()
    const input = wrapper.find('input')

    expect(wrapper).toMatchSnapshot()
    expect(input).toHaveLength(1)
    expect(input.get(0).props.type).toBe('text')
    expect(input.get(0).props.required).toBeTruthy()
    expect(input.get(0).props.placeholder).toBe('email')
  })
})
