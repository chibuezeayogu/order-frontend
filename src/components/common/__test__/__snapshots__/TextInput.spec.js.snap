// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component should render without crashing 1`] = `
<input
  className="form-control rounded-left"
  id="email"
  onChange={[MockFunction]}
  placeholder="email"
  required={true}
  type="text"
  value=""
/>
`;
